pkg/apis/core/validation/validation.go

```diff
@@ -5503,6 +5503,21 @@ func ValidatePodResize(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		allErrs = append(allErrs, field.Invalid(specPath, newPod.Status.QOSClass, "Pod QOS Class may not change as a result of resizing"))
 	}
 
+	// TODO(#128675): Do not allow removing resource requests/limits for Burstable Pods.
+	// TODO: Check for sidecar containers too.
+	for ix, ctr := range oldPod.Spec.Containers {
+		resourcesRemoved := func(resourceList, oldResourceList core.ResourceList) bool {
+			if len(oldResourceList) > len(resourceList) {
+				return true
+			}
+
+			return false
+		}
+		if resourcesRemoved(newPod.Spec.Containers[ix].Resources.Requests, ctr.Resources.Requests) || resourcesRemoved(newPod.Spec.Containers[ix].Resources.Limits, ctr.Resources.Limits) {
+			allErrs = append(allErrs, field.Forbidden(specPath, "resource requests and limits cannot be removed"))
```

Comment:
Probably good to give a more precise field path for this. At least separate the requests & limits cases.

Reply:
Done.